bedrock/newsletter/templates/newsletter/recovery.html

```diff
@@ -1,41 +1,73 @@
-{% extends 'newsletter/base.html' %}
+{# This Source Code Form is subject to the terms of the Mozilla Public
+  # License, v. 2.0. If a copy of the MPL was not distributed with this
+  # file, You can obtain one at http://mozilla.org/MPL/2.0/. -#}
+
+{% extends 'newsletter/base.html' %}}
+
+{% from "macros-protocol.html" import hero with context %}
 
 {% set_lang_files "mozorg/newsletters" %}
 
 {# "noindex" pages should not have the canonical or hreflang tags: bug 1442331 #}
 {% block canonical_urls %}<meta name="robots" content="noindex,follow">{% endblock %}
 
-{% block body_id %}newsletter-recovery{% endblock body_id %}
-
 {% block page_title %}{{ _('Newsletter email recovery') }}{% endblock page_title %}
 
 {% block content %}
+<main role="main" class="mzp-l-content">
   {% if form %}
-    {% block main_feature %}
-      <div id="main-feature">
-        <h1>{{ _('Manage your <span>Newsletter Subscriptions</span>') }}</h1>
-      </div>
-    {% endblock %}
-    <form method="post" action="{{ url('newsletter.recovery') }}" id="newsletter-recovery-form" class="container billboard">
-      {% if form.non_field_errors() %}
-        <div class="errorlist">
-          {{ form.non_field_errors() }}
-        </div>
-      {% endif %}
+    <div class="mzp-l-content mzp-t-narrow">
+      <h1>{{ _('Manage your <span>Newsletter Subscriptions</span>') }}</h1>
+
+      <form method="post" action="{{ url('newsletter.recovery') }}" id="newsletter-form" class="mzp-c-newsletter-form">
+        <header class="mzp-c-newsletter-header">
+          <p class="mzp-c-newsletter-tagline">{{ _('Enter your email address and we’ll send you a link to your email preference center.') }}</p>
+        </header>
 
-      <div class="form-fields">
-        <p class="intro">{{ _('Enter your email address and we’ll send you a link to your email preference center.') }}</p>
+        {% if form.non_field_errors() %}
+        <div class="mzp-c-form-errors" id="newsletter-errors">
+          {{ form.non_field_errors()|safe }}
+        </div>
+        {% endif %}
 
-        <div class="field email-field">
-          <div class="field-contents">
+        <fieldset class="mzp-c-newsletter-content">
+          <p>
             <label for="id_email">{{ _('Your email address:') }}</label>
-            {{ form.email }}
+            <input type="email" class="mzp-js-email-field" id="id_email" name="email" required aria-required="true">
             {{ form.email.errors }}
+          </p>
+
+          <div id="newsletter-details" class="mzp-c-newsletter-details">
+            <input type="hidden" name="lang" id="id_lang"  value="en" aria-required="false">
```

Comment:
I'm a little confused here maybe, but Is this needed for the email recovery form? It doesn't seem to be included on the [current page](https://www.mozilla.org/en-US/newsletter/recovery/) in production.

Reply:
When I try and complete the form, I see an error in the console:

```
TypeError: document.getElementById(...) is null
```

Might it be better to keep try and keep this simple, so that it doesn't try and behave like a newsletter?